src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,225 @@
+#  coding=utf-8
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import re
+from typing import Dict, List, Optional, Tuple, Union
+
+import pandas as pd
+import textdescriptives as td
+from tqdm import tqdm
+
+from argilla.client.feedback.dataset.local.dataset import FeedbackDataset
+from argilla.client.feedback.schemas.metadata import (
+    FloatMetadataProperty,
+    IntegerMetadataProperty,
+    TermsMetadataProperty,
+)
+from argilla.client.feedback.schemas.records import FeedbackRecord
+
+
+class TextDescriptivesExtractor:
+    """This class extracts a number of basic text descriptives from FeedbackDataset
+    records using the TextDescriptives library and adds them as record metadata."""
+
+    def __init__(
+        self,
+        model: str = "en",
+        metrics: list[str] = ["quality", "readability", "information_theory"],
```

Comment:
can we also define the sub-metrics directly.

Reply:
@davidberenstein1957 Do we want to offer the user the opportunity to use all the metrics and submetrics available in text-descriptives and only suggest basic/default metrics to them?

Reply:
Ideally yes, but I realized there is not very clear hierarchy of metrics and sub metrics available so we need to think about a convenient way to implement this.

Reply:
If we choose to let the user instantiate by defining their desired submetrics (as opposed to metrics), they will have 75 submetrics to choose from. Given the magnitude of available submetrics, do we still want to allow the user to define them?

Reply:
Yes, I think we should provide the defaults. `metrics` and `sub_metrics` and let the users fine-tune whenever needed.

Reply:
I think we need to hold off on allowing the user to specify submetrics for right now, as described in previous comment.